From 8027ca7c6db101e17f4737834c9ba31e0e905631 Mon Sep 17 00:00:00 2001
From: Herman van Hazendonk <github.com@herrie.org>
Date: Wed, 6 Mar 2019 16:38:32 +0100
Subject: [PATCH] rules: consider MMC device partitions with partition number >
 9 & create disk/by-partlabel links for mmcblk partitions

Add entries for extra partitions found on MMC devices (common in Chromebooks).

Create disk/by-partlabel links for mmcblk partitions

Android based kernel exports the uevent property PARTNAME, which can be
used to find out at run time the named partitions (e.g. boot) for the
device.

This is specially useful for the Touch based images and flash-kernel,
to automatically update the kernel by writing at the correct partition
(independently of the hardware revision).

Signed-off-by: Martin Jansa <Martin.Jansa@gmail.com>
Signed-off-by: Herman van Hazendonk <github.com@herrie.org>
Upstream-Status: Pending
---
 rules.d/60-persistent-storage.rules | 1 +
 1 file changed, 1 insertion(+)

diff --git a/rules.d/60-persistent-storage.rules b/rules.d/60-persistent-storage.rules
index ad5cca0aef..b437b3bbc5 100644
--- a/rules.d/60-persistent-storage.rules
+++ b/rules.d/60-persistent-storage.rules
@@ -94,6 +94,7 @@ KERNEL=="mmcblk[0-9]", SUBSYSTEMS=="mmc", ATTRS{serial}=="?*", ENV{ID_SERIAL}="$
 KERNEL=="mmcblk[0-9]", SUBSYSTEMS=="mmc", ATTRS{name}=="?*", ENV{ID_NAME}="$attr{name}"
 KERNEL=="mmcblk[0-9]", ENV{ID_NAME}=="?*", ENV{ID_SERIAL}=="?*", SYMLINK+="disk/by-id/mmc-$env{ID_NAME}_$env{ID_SERIAL}"
 KERNEL=="mmcblk[0-9]p[0-9]*", ENV{ID_NAME}=="?*", ENV{ID_SERIAL}=="?*", SYMLINK+="disk/by-id/mmc-$env{ID_NAME}_$env{ID_SERIAL}-part%n"
+KERNEL=="mmcblk[0-9]p[0-9]*", ENV{PARTNAME}=="?*", SYMLINK+="disk/by-partlabel/$env{PARTNAME}"
 
 # Memstick
 KERNEL=="msblk[0-9]|mspblk[0-9]", SUBSYSTEMS=="memstick", ATTRS{name}=="?*", ATTRS{serial}=="?*", \
